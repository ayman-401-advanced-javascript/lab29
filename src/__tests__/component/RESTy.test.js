/* eslint-disable no-undefined */
/* eslint-disable no-unused-vars */
import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Form from '../../component/form/form.js';
import Results from '../../component/results/results.js';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<Form />', ()=>{
  it('exists at the start of our application',()=>{
    let app = shallow(<Form />);
    expect(app.find('button').exists()).toBeTruthy();
    expect(app.find('input').exists()).toBeTruthy();
    expect(app.find('form').exists()).toBeTruthy();

  });
  it('GET Method',()=>{
    let app = mount(<Form />);
    let button = app.find('button');
    button.simulate('click');
    expect(app.state('Headers')).toBe(undefined);
    expect(app.state('Response')).toBe(undefined);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});