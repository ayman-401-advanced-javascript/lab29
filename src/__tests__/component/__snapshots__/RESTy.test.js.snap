// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="useInput"
    name="url"
    onChange={[Function]}
    placeholder="Type the URL  for the API"
    type="text"
  />
  <br />
  <label>
    GET
    <input
      name="GET"
      required={true}
      type="radio"
    />
  </label>
  <label>
    POST
    <input
      name="POST"
      type="radio"
    />
  </label>
  <label>
     PUT
    <input
      name="PUT"
      type="radio"
    />
  </label>
  <label>
    DELETE
    <input
      name="DELETE"
      type="radio"
    />
  </label>
  <button
    type="submit"
  >
    Go!
  </button>
</form>
`;
